hw/top_earlgrey/syn/chip_earlgrey_asic.sdc

```diff
@@ -90,27 +90,20 @@ set IO_TCK_PERIOD [expr $IO_TCK_TARGET_PERIOD*$CLK_PERIOD_FACTOR]
 create_clock -name IO_CLK -period ${IO_TCK_PERIOD} [get_pins ${IO_CLK_PIN}]
 set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} [get_clocks IO_CLK]
 
-# This requires knowledge of actual pin names, which are different depending on
-# whether we run this with tech libs or not.
-if {$FOUNDRY_ROOT != ""} {
-    set CLK_MUX_SRC_NAME u_size_only_mux2/D1
-    set CLK_MUX_DST_NAME u_size_only_inv/${DRIVING_CELL_PIN}
-} else {
-    # this is only used by SDC on generic views
-    set CLK_MUX_SRC_NAME clk0_i
-    set CLK_MUX_DST_NAME clk_o
-}
+# This requires knowledge of actual port name
+set CLK_SRC_NAME clk_i
+set CLK_DST_NAME clk_o
 
 # generated clocks (div2/div4)
-set CLK_MUX_PATH top_earlgrey/u_clkmgr_aon/u_no_scan_io_div2_div/u_clk_mux/gen_*u_impl*
+set CLK_PATH top_earlgrey/u_clkmgr_aon/u_no_scan_io_div2_div
```

Comment:
could you check in with @msfschaffner @eunchan and run a quick synth to make sure this is okay?

Reply:
Sounds good.
@msfschaffner @eunchan can you give me instruction how to run a syn?

Reply:
Synthesis result seems to be the same as the result in the master branch.

Reply:
@arnonsha can you review this fix? Maybe you can run synthesis on your side to check if anything can affect.

Reply:
One question I have here is whether this behaves correctly post-CTS insertion (I don't know).

As long as the input clock net is ideal and has no prop-delays, it should not matter whether we take `IO_CLK_PIN` as a reference here. However, once we are post-CTS, I think it may matter and I believe this constraint has been formulated with that in mind.

@arnonsha what is your preferred format here?

Reply:
i'm not sure i completely understand the concern, are you saying this could be a problem because it's not at the leaf node but at a wrapper layer above?

Reply:
Yes because the leaf node will eventually have a propagation delay with respect to the clock root, and timing analysis should take that into account for the generated clock as well... although the create_generated_clock command may already compensate for source latency, in which case we are ok. Just wanted to double check this is handled correctly.

Reply:
Thanks for comments. Yes, the network latency will be handled by create_generated_clock.

Reply:
Thanks for confirming - in that case this LGTM.